src/assets/js/protocol/protocol-utils.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+// create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+(function() {
+    'use strict';
+
+    var Utils = {};
+
+    // matches() polyfill
+    // TODO: where should this go?
+    if (!Element.prototype.matches) {
+        Element.prototype.matches = Element.prototype.msMatchesSelector || Element.prototype.webkitMatchesSelector;
+    }
+
+    /**
+     * wrapAll
+     * @param {Object} el - Element that you want to get the siblings of
+     * @param {String} selector - Selector of the element you want to stop collecting siblings before
+     * @param {String} filter - Selector to filter results if you only want siblings matching this pattern
+     * - https://gomakethings.com/how-to-get-all-sibling-elements-until-a-match-is-found-with-vanilla-javascript/
+     */
+    Utils.nextUntil = function (el, selector, filter) { //TODO: captialization of nextUntil?
+        // Setup siblings array
+        var siblings = [];
+
+        // Get the next sibling element
+        el = el.nextElementSibling;
+
+        // As long as a sibling exists
+        while (el) {
+
+            // If we've reached our match, bail
+            if (el.matches(selector)) break;
+
+            // If filtering by a selector, check if the sibling matches
+            if (filter && !el.matches(filter)) {
+                el = el.nextElementSibling;
+                continue;
+            }
+
+            // Otherwise, push it to the siblings array
+            siblings.push(el);
+
+            // Get the next sibling element
+            el = el.nextElementSibling;
+        }
+
+        return siblings;
+    };
+
+    window.Mozilla.Utils = Utils;
```

Comment:
Oh, here's something we haven't thought through quite yet. It's probably only a problem on bedrock, but we already have a `Mozilla.Utils` namespace there :(

We should maybe file an issue and think about namespacing our protocol JS to something more unique `MZP.`?

For now though, maybe we could use `DOMUtils`?

Reply:
Oh we should just be using a MZP namespace globally.

Reply:
Damn that's going to take some changing. I'll add an issue, we can discuss it on Thursday.

Reply:
I'm using `mzp` everywhere now too.